cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -183,13 +186,28 @@ public final ProgramController run(final Program program, final ProgramOptions o
       final ProgramOptions options = addArtifactPluginFiles(oldOptions, localizeResources,
                                                             DirUtils.createTempDir(tempDir));
 
+      final List<String> additionalClassPaths = new ArrayList<>();
+      List<String> newExtraJars = new ArrayList<>();
```

Comment:
they will replace the existing jars in cConf, maybe to rename it to `cConfNewExtraJars`?